_posts/2021-07-08-backup-restore-che-installation.md

```diff
@@ -0,0 +1,243 @@
+    awsAccessKeySecretRef: aws-user-credentials-secret
+```
+Both secrets `aws-backup-encryption-password-secret` with `repo-password` key and `aws-user-credentials-secret` with `awsAccessKeyId` and `awsSecretAccessKey` keys must exist.
+
+As it was described above, under `spec` section only `rest`, `awss3` and `sftp` is allowed.
+CR definitions have self-explanatory fields and it will be easy to create a backup server configuration.
+But note, that each subsection mutually excludes the others.
+However, it is allowed to create as many backup server configurations as needed.
+
+#### Backing up
+
+To create a new backup, a new CR of `CheClusterBackup` type should be created:
+```yaml
+apiVersion: org.eclipse.che/v1
+kind: CheClusterBackup
+metadata:
+  name: eclipse-che-backup
+spec:
+  backupServerConfigRef: backup-server-configuration
+```
+Right after the CR creation a new backup process will be started.
+To monitor backup process state, one should look at `status` section of the created CR:
+```bash
+$ kubectl get CheClusterBackup eclipse-che-backup -n eclipse-che -o yaml | grep -A 5 ^status
+```
+The output of the command above looks like:
+```yaml
+status:
+  message: 'Backup is in progress. Start time: <timestamp>'
+  stage: Collecting Che installation data
+  state: InProgress
+```
+where
+* `message` shows overall human readable status or an error message.
+* `stage` displays human readable current phase of backup process
+* `state` indicates the overall state of the backup. Only `InProgress`, `Succeeded` and `Failed` allowed.
+
+When the process finishes successfully, the `status` section will contain `snapshotId` field that could be used when restoring.
+The CR might be deleted after backup is finished.
+
+If one need to request internal backup server and create a backup, `CheClusterBackup` with `useInternalBackupServer` property set to `true` should be created:
+```yaml
+apiVersion: org.eclipse.che/v1
+kind: CheClusterBackup
+metadata:
+  name: eclipse-che-backup
+spec:
+  useInternalBackupServer: true
+```
+Note, it will create an instance of `CheBackupServerConfiguration` and corresponding secrets automatically.
+
+#### Restoring
+
+To restore from a backup snapshot, a new CR of `CheClusterRestore` type should be created:
+```yaml
+apiVersion: org.eclipse.che/v1
+kind: CheClusterRestore
+metadata:
+  name: eclipse-che-restore
+spec:
+  backupServerConfigRef: backup-server-configuration
+```
+
+By default the latest snapshot is taken.
+However, it is possible to restore from a specific snapshot by adding `snapshotId` field under `spec` section.
+
+To monitor the restore state, one may read status of the corresponding CR:
+```bash
+$ kubectl get CheClusterRestore eclipse-che-restore -n eclipse-che -o yaml | grep -A 5 ^status
+```
+Once the restore finishes, the CR can be deleted.
+
+### Limitations
+
+As of now, there are two major limitations with backup and restore:
```

Comment:
Are there some issues to track these limitations? It would be nice to put a link to the GH issues here.

Reply:
> Backing up of user's projects inside workspaces hasn't been implemented yet. So, all not committed changes will not be restored.

There is no separate issue yet. An item for the problem present in [this epic](https://github.com/eclipse/che/issues/18043).

> Backup snapshots are bind to the specific cluster, so it is not possible to restore snapshot on another cluster in general case. This is because Che binds to some cluster ID's.

The problem was discussed on cabal call. So far no plans to fix as Che users are bound to cluster users.